src/NewTwitchApi/RequestResponse.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NewTwitchApi;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class RequestResponse
```

Comment:
Why do we need this? API consumer does not need to know about request, that's why he is using this library to pass parameters, and get the response. He adds dependency to get rid of messing with the request. He don't need to bother with HTTP, request etc.

Reply:
So, initially, there were times when I wanted access to the Request as well as the Response. One instance of this still remains in the CLI client, here:

https://github.com/ChessCom/twitch-api-php/blob/3306e8e98d401fa02969bc078aebcbed26b69c80/src/NewTwitchApi/Cli/CliClient.php#L74

I use that the Request given to me in RequestResponse to echo the the URI that was called (very useful information in the CLI client output).

That said, I _think_ that's the only time I actually use Request from RequestResponse. If you can help me get the Request info displaying there in a different way that doesn't use RequestResponse, I wouldn't mind getting rid of it.

Reply:
Yes, inject guzzle and https://michaelstivala.com/logging-guzzle-requests/. ;)